File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file141.f90
program DRB091_threadprivate2_orig_no
    use omp_lib
    use DRB091
    implicit none

    integer :: len, i, sum
    len = 1000
    sum = 0

    !$omp parallel copyin(sum0)
        !$omp do
        do i = 1, len
            sum0 = sum0+i
        end do
        !$omp end do
        !$omp critical
        sum = sum+sum0
        !$omp end critical
    !$omp end parallel

    do i = 1, len
        sum1 = sum1+i
    end do

    print*,'sum =',sum,'sum1 =',sum1

end program

File: data/gpt-4o/readable_format_HPEC/cpp/file141.cpp
#include <stdio.h>
#include <omp.h>

int main()
{
    int len = 1000;
    int sum = 0;
    int sum0 = 0;

#pragma omp threadprivate(sum0)

#pragma omp parallel copyin(sum0)
    {
#pragma omp for
        for(int i = 0; i < len; i++)
        {
            sum0 += i+1;
        }

#pragma omp critical
        sum += sum0;
    }

    int sum1 = 0;
    for(int i = 0; i < len; i++)
    {
        sum1 += i+1;
    }

    printf("sum = %d, sum1 = %d
", sum, sum1);

    return 0;
}
